### PR TITLE
[cli] Invalidate build match for changes to `config` in builds

### PR DESCRIPTION
**Pull request by TooTallNate**
* Fixes changing a `config` value in the `builds` array (such as `helpers: false` for `@vercel/node`) and having it be reflected in `vc dev` without restarting the dev server.
* Re-validates the env vars configuration when a `.env` file changes or the `env` object is changed in `vercel.json` (same for the builds equivalents).
* Ensures that the `NODEJS_HELPERS` build env var is being properly checked in `startDevServer()`.
* Regenerated the `yarn.lock` file because yarn was erroring when trying to add the `fast-deep-equal` dependency.

### CODE REVIEW

File: packages/frameworks/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "devDependencies": {
     "@types/jest": "24.0.22",
     "@types/node": "12.0.4",
-    "ajv": "6.10.2",
+    "ajv": "6.12.2",
```
Comment (TooTallNate) [inline packages/frameworks/package.json:12]: This update was to remove a duplicated `ajv` dependency in `yarn.lock`.

File: packages/now-cli/test/dev/fixtures/node-helpers/vercel.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "builds": [{ "src": "index.js", "use": "@vercel/node@canary" }] }
```
Comment (styfle) [inline packages/now-cli/test/dev/fixtures/node-helpers/vercel.json:1]: Is this file necessary?

File: packages/now-cli/src/util/dev/server.ts [hunk 3]
```diff
@@ -2261,3 +2272,11 @@ function hasNewRoutingProperties(nowConfig: NowConfig) {
     typeof nowConfig.trailingSlash !== undefined
   );
 }
+
+function buildMatchEquals(a?: BuildMatch, b?: BuildMatch): boolean {
+  if (!a || !b) return false;
+  if (a.src !== b.src) return false;
+  if (a.use !== b.use) return false;
+  if (!deepEqual(a.config || {}, b.config || {})) return false;
+  return true;
```
Comment (styfle) [inline packages/now-cli/src/util/dev/server.ts:2281]: Is there a reason why you are only comparing 3 properties (src, use, config) instead of all?

```ts
function buildMatchEquals(a?: BuildMatch, b?: BuildMatch)
  return deepEqual(a, b);
}
```

File: packages/now-cli/test/dev/fixtures/node-helpers/vercel.json (hunk 2 above)
Comment (styfle) [reply packages/now-cli/test/dev/fixtures/node-helpers/vercel.json:1]: Nevermind, I see it is used in the test below 😄

**Review by styfle — APPROVED** [review]

File: packages/now-cli/src/util/dev/server.ts (hunk 3 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/server.ts:2281]: I guess there are a lot of properties on `BuildMatch` and we only want to compare the properties on `BuildConfig` 👍

**Review by AndyBitz — APPROVED** [review]
